### DISCUSSION TITLE
GUI Data

### DISCUSSION THREAD

**Discussion by 1madison24** [OpenEVSE/openevse_esp32_firmware#777] [Q&A]
Hello, 

I am trying to manipulate how power receives data, thee issue is I cannot find how the gui receives data from the host codebase. Where is this? 

Thanks!

**Comment by jeremypoulter** [comment]
The GUI retrieves the initial data via the [HTTP API](https://openevse.stoplight.io/docs/openevse-wifi-v4/6e3b6e2f1b976-open-evse-wi-fi-for-esp-32) then after that updates are feed via a WebSocket,
